// navcm/index.js
exports.navCM=function() {
    var nav = {};
    nav.json = [];

    nav.get_hierarchy_json = function (data, id) {
        if (!id) {
            id = 0
        }
        function hierarchy(data, ids, cenji) {
            for (var i = 0; i < data.length; i++) {
                if (data[i].pid == ids) {
                    for (var ii = 0; ii <= cenji; ii++) {
                        if (ii == 0) {
                            data[i].limit = ""
                        }
                        else if (ii == 1) {
                            data[i].limit = "|—"
                        } else if (ii > 1) {
                            data[i].limit += "—";
                        }
                    }
                    data[i].hierarchy = cenji;
                    nav.json.push(data[i]);
                    hierarchy(data, data[i].id, cenji + 1);
                }
            }
        }

        hierarchy(data, id, 0);
        return this.json;
    }

    nav.get_hierarchy_ul = function (data, id) {
        function treeMenu(a) {
            this.tree = a || [];
            this.groups = {};
        };
        treeMenu.prototype = {
            init: function (pid) {
                if (!pid) {
                    pid = 0
                }
                this.group();
                return this.getDom(this.groups[pid]);
            },
            group: function () {
                for (var i = 0; i < this.tree.length; i++) {
                    if (this.groups[this.tree[i].pid]) {
                        this.groups[this.tree[i].pid].push(this.tree[i]);
                    } else {
                        this.groups[this.tree[i].pid] = [];
                        this.groups[this.tree[i].pid].push(this.tree[i]);
                    }
                }
            },
            getDom: function (a) {
                if (!a) {
                    return "";
                }
                var html = '\n<ul>\n';
                for (var i = 0; i < a.length; i++) {
                    html += '<li data-id="' + a[i].id + '">';
                    for (var key in a[i]) {
                        html += '<span class="' + key + '">' + a[i][key] + '</span>'
                    }
                    html += '<span></span>';
                    html += this.getDom(this.groups[a[i].id]);
                    html += '</li>\n';
                }
                ;
                html += '</ul>\n';
                return html;
            }
        };
        var html = new treeMenu(data).init(id);
        return html;
    }

    nav.get_hierarchy_table = function (data, id) {
        if (!id) {
            id = 0
        }
        var html = "";

        function hierarchy(data, ids, cenji) {
            for (var i = 0; i < data.length; i++) {
                if (data[i].pid == ids) {
                    html += '<div class="tr" data-id="' + data[i].id + '" data-pid="' + data[i].pid + '">';
                    for (var key in data[i]) {
                        if (key == "navName") {
                            html += '<div class="td" style="text-indent:' + 20 * cenji + 'px">' + data[i][key] + '</div>'
                        } else if (key != "pid") {
                            html += '<div class="td">' + data[i][key] + '</div>'
                        }
                    }
                    html += '<div class="td update" data-id="' + data[i].id + '" data-pid="' + data[i].pid + '">编辑</div>';
                    html += '<div class="td remove" data-id="' + data[i].id + '" data-pid="' + data[i].pid + '">删除</div>';
                    html += "</div>"
                    hierarchy(data, data[i].id, cenji + 1);
                }
            }
        }

        hierarchy(data, id, 0);
        return html;
    }
    nav.get_hierarchy_select = function (data, id) {
        if (!id) {
            id = 0
        }
        var html = "";
        var str = "";

        function hierarchy(data, ids, cenji) {
            for (var i = 0; i < data.length; i++) {
                if (data[i].pid == ids) {
                    for (var ii = 0; ii <= cenji; ii++) {
                        if (ii == 0) {
                            data[i].limit = ""
                        }
                        else if (ii == 1) {
                            data[i].limit = "|—"
                        } else if (ii > 1) {
                            data[i].limit += "—";
                        }
                    }
                    html += '<option class="item"  data-id="' + data[i].id + '" data-pid="' + data[i].pid + '" data-Name="' + data[i].navName + '" value="' + data[i].id + '">' + data[i].limit + data[i].navName + '</option>';
                    hierarchy(data, data[i].id, cenji + 1);
                }
            }
        }

        hierarchy(data, id, 0);
        return html;
    }
    nav.get_insertAll_id = function (Da, id) {
        var IDs = [];
        IDs.push(id);
        function hierarchy(data, ids) {
            for (var i = 0; i < data.length; i++) {
                if (data[i].pid == ids) {
                    IDs.push(data[i].id)
                    hierarchy(data, data[i].id);
                }
            }
        }

        hierarchy(Da, id);
        return IDs.join(",");
    }
    nav.get_parent = function (Da, pid) {
        var IDs = [];

        function hierarchy(data, pids) {
            for (var i = 0; i < data.length; i++) {
                if (data[i].id == pids) {
                    IDs.push(data[i])
                    hierarchy(data, data[i].pid);
                }
            }
        }

        hierarchy(Da, pid);
        return IDs;
    }
    return nav;
}
